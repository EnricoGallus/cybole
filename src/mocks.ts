export function registerElectron() {
    const electronMock = {
        openDialog: jest.fn().mockReturnValue(Promise.resolve({ filePaths: ['directory'] })),
        getFiles: jest.fn(),
        readFile: jest.fn(),
        writeFile: jest.fn(),
        getAppDescription: jest.fn().mockReturnValue(Promise.resolve('Cybole v0.2')),
        openContextMenu: jest.fn(),
        on: jest.fn(),
    };

    window.electron = electronMock;
    return electronMock;
}
