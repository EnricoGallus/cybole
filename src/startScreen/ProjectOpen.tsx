import {useState} from "react";
import {useNavigate} from "react-router-dom";
import {Button, Grid, Input} from "@geist-ui/core";

function ProjectOpen() {
    const [directory, setDirectory] = useState('');
    const navigate = useNavigate();

    return (
        <Grid.Container>
            <Grid xs={24}>
                Project New
            </Grid>
            <Grid xs={24}>
                <Input aria-label="project-directory-input" value={directory} readOnly onClick={() => {
                    let dialogConfig = {
                        title: 'Select the Directory of the project',
                        buttonLabel: 'Select Directory',
                        properties: ['openDirectory']
                    };
                    window.electron.openDialog('showOpenDialog', dialogConfig)
                        .then(result => setDirectory(result.filePaths[0]));
                }} />
                <Button onClick={() => {navigate('/editor', { state: { directory: directory}})}}>Open Project</Button>
            </Grid>
        </Grid.Container>
    )
}

export default ProjectOpen;