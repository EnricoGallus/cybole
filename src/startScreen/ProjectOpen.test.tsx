import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import ProjectOpen from './ProjectOpen';
import {registerElectron} from "../tests/mocks";

describe('ProjectOpen', () => {
    test('renders component', () => {
        const { container } = render(<ProjectOpen />, { wrapper: MemoryRouter });

        expect(container).toMatchSnapshot();
    });

    test('openDirectoryDialog should set selected directory to input', async () => {
        let electronMock = registerElectron();
        render(<ProjectOpen />, { wrapper: MemoryRouter });

        const input = screen.getByLabelText('project-directory-input');

        fireEvent.click(input);

        await waitFor(() =>
            expect(electronMock.openDialog).toBeCalledWith('showOpenDialog', {
                title: 'Select the Directory of the project',
                buttonLabel: 'Select Directory',
                properties: ['openDirectory'],
            })
        );

        await waitFor(() => expect(input).toHaveValue('directory'));
    });
});
