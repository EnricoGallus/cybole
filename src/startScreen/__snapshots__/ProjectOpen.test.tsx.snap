// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectOpen renders component 1`] = `
<div>
  <div
    class="jsx-3633143255 item jsx-478091869"
  >
    <div
      class="jsx-3066557810 item xs jsx-3484918754"
    >
      Project New
    </div>
    <div
      class="jsx-3066557810 item xs jsx-3484918754"
    >
      <div
        class="jsx-599062270 with-label"
      >
        <div
          class="jsx-599062270 input-container"
        >
          <div
            class="jsx-599062270 input-wrapper"
          >
            <input
              aria-label="project-directory-input"
              autocomplete="off"
              class="jsx-599062270 "
              placeholder=""
              readonly=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <button
        class="jsx-3108082297 btn"
        type="button"
      >
        <div
          class="text"
        >
          Open Project
        </div>
      </button>
    </div>
  </div>
</div>
`;
